build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/rules/AbstractGradleViolationRule.groovy

```diff
@@ -42,9 +42,17 @@ import javax.inject.Inject
 abstract class AbstractGradleViolationRule extends AbstractContextAwareViolationRule {
 
     private final Map<ApiChange, String> acceptedApiChanges
+    private final File apiChangesJsonFile
+    private final File projectRootDir
 
     AbstractGradleViolationRule(Map<String, String> acceptedApiChanges) {
-        this.acceptedApiChanges = AcceptedApiChanges.fromAcceptedChangesMap(acceptedApiChanges)
+        this(acceptedApiChanges, null, null)
+    }
+
+    AbstractGradleViolationRule(Map<String, String> acceptedApiChanges, String apiChangesJsonFile, String projectRootDir) {
```

Comment:
In order to avoid a cast everywhere else, consider having this constructor accept a `Map<String, Object>`

Reply:
I am torn on this change, because in reality, it is only a `Map<String, String>` (per `RichReport#addRule`). Perhaps we can change this after a PR to upstream?